src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: object[], key: string): GroupsMap<object> {
+  const objectResult = {} as GroupsMap<object>;
```

Comment:
```suggestion
  const objectResult = {};
```
try to avoid [Type Assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) and `object` type
just accept type in function as a generic and use